### PR TITLE
Convert JsChecker WARNINGS to ERRORS

### LINKED ISSUE #144: pedantic mode for closure_js_library()
IMO it would be nice to add an (optional) pedantic mode to `closure_js_library()` which converted all WARNINGS to ERRORS.

```
  private static final ImmutableSet<String> WARNINGS =
      ImmutableSet.of(
          "checkTypes",
          "deprecated",
          "deprecatedAnnotations",
          "extraRequire",
          "lintChecks",
          "nonStandardJsDocs");
```

@jart I can't remember, have we talked about this before? Would you be open to such a PR?

### PR DESCRIPTION
**Pull request by hochhaus**
Fixed #144

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: java/com/google/javascript/jscomp/JsChecker.java [hunk 1]
```diff
@@ -66,20 +66,16 @@
...
 
-  private static final ImmutableSet<String> WARNINGS =
+  private static final ImmutableSet<String> ERRORS =
       ImmutableSet.of(
+          "checkRegExp",
           "checkTypes",
           "deprecated",
           "deprecatedAnnotations",
           "extraRequire",
           "lintChecks",
```
Comment (jart) [inline java/com/google/javascript/jscomp/JsChecker.java]: Nit: Please sort the items in this list.

Comment (hochhaus) [inline java/com/google/javascript/jscomp/JsChecker.java]: I'm missing it... What isn't sorted? These items appear alphabetized to me.

Comment (jart) [inline java/com/google/javascript/jscomp/JsChecker.java]: My mistake. The diff confused me.

**Comment by jart** [comment]
LGTM. Thanks for doing this.

Mr. Jenkins: test this please.

**Pull request merged**